src/NativeTypes/Drawing.tt

```diff
@@ -50,7 +50,8 @@ namespace CoreGraphics
 
 		public static bool operator == (<#= type.Name #> l, <#= type.Name #> r)
 		{
-			return l.<#= type.X_Width #> == r.<#= type.X_Width #> && l.<#= type.Y_Height #> == r.<#= type.Y_Height #>;
+			// the following version of Equals cannot be removed by the linker, while == can be
+			return l.Equals (r);
```

Comment:
There's a performance hit here: you're calling a virtual method now to do the comparison (so vtable lookup + method invocation) instead of this static method (which the JIT/AOT compiler might even inline in the caller).

I'm not sure how much of a hit this would be though, it could be insignificant.

Reply:
Hmm.. unlike `Equals(Object)` [0], `Equals(T) is not explicitly virtual, but it does map to an interface.
Also since they are `struct`, which cannot be subclassed, I would think it's inline-able...
Still a good point, I'll look at the sharplab's release assembly (even if it won't be mono/llvm aot output)

[0] which gets `sealed` in release builds (at least for legacy right now, still on todo (but I think I know how now) for net6

Reply:
@rolfbjarne https://gist.github.com/spouliot/e0f462435baa5a6389d39db8d41bd65e

JIT'ed code is identical, so it is (or could/should be) inlined